apps/examples/wifi_manager/wm_test/wm_test_connectbyrssi_test.c

```diff
@@ -0,0 +1,248 @@
+	do {                            \
+		WT_LOG(TAG, "wait signal"); \
+		sem_wait(&g_wm_sem);       \
+	} while (0)
+
+
+static void _wt_sta_connected(wifi_manager_cb_msg_s msg, void *arg);
+static void _wt_sta_disconnected(wifi_manager_cb_msg_s msg, void *arg);
+static void _wt_scan_done(wifi_manager_cb_msg_s msg, void *arg);
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	_wt_sta_connected,
+	_wt_sta_disconnected,
+	NULL,
+	NULL,
+	_wt_scan_done,
+};
+
+void _wt_sta_connected(wifi_manager_cb_msg_s msg, void *arg)
+{
+	WT_LOG(TAG, "-->res(%d)", msg.res);
+	WT_LOG(TAG, "bssid %02x:%02x:%02x:%02x:%02x:%02x",
+		msg.bssid[0], msg.bssid[1],
+		msg.bssid[2], msg.bssid[3],
+		msg.bssid[4], msg.bssid[5]);
+
+	if (WIFI_MANAGER_SUCCESS == msg.res) {
+		WT_LOG(TAG, "Station Connected");
+		snprintf(g_connected_ap_mac, sizeof(g_connected_ap_mac), "%02x:%02x:%02x:%02x:%02x:%02x",
+			msg.bssid[0], msg.bssid[1],
+			msg.bssid[2], msg.bssid[3],
+			msg.bssid[4], msg.bssid[5]);
+	} else {
+		WT_LOGE(TAG, "Station Not Connected");
+	}
+	WM_TEST_SIGNAL;
+}
+
+void _wt_sta_disconnected(wifi_manager_cb_msg_s msg, void *arg)
+{
+	WT_LOG(TAG, "--> res(%d) reason %d", msg.res, msg.reason);
+	WM_TEST_SIGNAL;
+}
+
+void _wt_scan_done(wifi_manager_cb_msg_s msg, void *arg)
+{
+	WT_LOG(TAG, "-->");
+	/* Make sure you copy the scan results onto a local data structure.
+	 * It will be deleted as soon as you exit this function.
+	 */
+	if (msg.res != WIFI_MANAGER_SUCCESS || msg.scanlist == NULL) {
+		WM_TEST_SIGNAL;
+		return;
+	}
+	wt_print_scanlist(msg.scanlist);
+	wifi_manager_scan_info_s *wifi_scan_iter = msg.scanlist;
+	while (wifi_scan_iter != NULL) {
+		if (strncmp(wifi_scan_iter->ssid, opt->ssid, WIFIMGR_SSID_LEN) == 0) {
+			strncpy(g_bssid, wifi_scan_iter->bssid, strlen(wifi_scan_iter->bssid));
+			break;
+		}
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+	WM_TEST_SIGNAL;
+}
+
+static int wm_connect_by_rssi_init(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	res = wifi_manager_init(&g_wifi_callbacks);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to init wifi_manager");
+		return 1;
+	}
+	WT_LEAVE;
+	return 0;
+}
+
+static void wm_connect_by_rssi_deinit(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = wifi_manager_deinit();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "WiFi Manager failed to stop");
+	}
+	WT_LEAVE;
+}
+
+static void _wt_scan(void *arg)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	res = wifi_manager_scan_ap(NULL);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to scan");
+		return;
+	}
+	WM_TEST_WAIT; // wait the scan result
+	WT_LEAVE;
+}
+
+static void _wt_connect(void *arg)
+{
+	WT_ENTER;
+	struct wt_options *ap_info = (struct wt_options *)arg;
+	wifi_manager_ap_config_s apconfig;
+	strncpy(apconfig.ssid, ap_info->ssid, WIFIMGR_SSID_LEN);
+	apconfig.ssid_length = strlen(ap_info->ssid);
+	apconfig.ssid[WIFIMGR_SSID_LEN] = '\0';
```

Comment:
it is not required. already WIFIMGR_SSID_LEN is copied

Reply:
Done.